pkg/controller/installation/core_controller.go

```diff
@@ -1783,3 +1779,35 @@ func removeInstallationFinalizer(i *operator.Installation) {
 		i.SetFinalizers(stringsutil.RemoveStringInSlice(CalicoFinalizer, i.GetFinalizers()))
 	}
 }
+
+func (r *ReconcileInstallation) updateInstallationDegraded(reason string, err error, instance *operator.Installation, ctx context.Context, log logr.Logger) {
+	r.SetDegraded(reason, err, log)
+	conditions := []operator.TigeraStatusCondition{
+		{Type: operator.ComponentDegraded, Status: operator.ConditionTrue, Reason: reason, Message: err.Error()},
```

Comment:
It is probably best to have a nil check before you call err.Error(). Maybe there are places in the code where err is passed as nil.